lib/file-view.coffee

```diff
@@ -19,7 +19,11 @@ class FileView extends HTMLElement
     @fileName.dataset.name = @file.name
     @fileName.dataset.path = @file.path
 
-    @fileName.classList.add(FileIcons.getService().iconClassForPath(@file.path))
+    iconClass = FileIcons.getService().iconClassForPath(@file.path, this)
```

Comment:
It allows the provider to understand the context of where the icon is being requested. To give a concrete example, [File-Icons](https://github.com/DanBrooker/file-icons) has a *"Show in tabs"* setting that should determine whether an icon gets displayed in a tab or not.